pkg/fill/service.go

```diff
@@ -0,0 +1,185 @@
+// VulcanizeDB
+// Copyright © 2022 Vulcanize
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package fill
+
+import (
+	"math"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/ethereum/go-ethereum/statediff"
+	"github.com/jmoiron/sqlx"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/vulcanize/ipld-eth-server/pkg/serve"
+)
+
+// WatchedAddress type is used to process currently watched addresses
+type WatchedAddress struct {
+	Address      string `db:"address"`
+	CreatedAt    uint64 `db:"created_at"`
+	WatchedAt    uint64 `db:"watched_at"`
+	LastFilledAt uint64 `db:"last_filled_at"`
+
+	StartBlock uint64
+	EndBlock   uint64
+}
+
+// Service is the underlying struct for the watched address gap filling service
+type Service struct {
+	db       *sqlx.DB
+	client   *rpc.Client
+	interval int
+}
+
+// NewServer creates a new Service
+func New(config *serve.Config) *Service {
+	return &Service{
+		db:       config.DB,
+		client:   config.Client,
+		interval: config.WatchedAddressGapFillInterval,
+	}
+}
+
+// Start is used to begin the service
+func (s *Service) Start() {
```

Comment:
If this is a standalone background process that has no server component and is disconnected from the rest of ipld-eth-server I'm not sure it belongs in this repo.